DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1

```diff
@@ -0,0 +1,268 @@
+                }
+            }
+            
+            if ($Quiet -or $isValid)
+            {
+                $isValid
+            }
+            else
+            {
+                throw [System.ArgumentException]"'$hash' is not a valid hash."
+            }
+        }
+    }
+}
+
+function Get-TargetResource 
+{
+    [CmdletBinding()]
+    [OutputType([Hashtable])]
+    param(
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-Thumbprint
+        } )]
+        [String]
+        $Thumbprint ,
+
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-CertificatePath
+        } )]
+        [String]
+        $Path ,
+
+        [Parameter()]
+        [ValidateSet(
+             'LocalMachine'
+        )]
+        [String]
+        $Location = 'LocalMachine' ,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Store = 'My'
+    )
+
+    @{
+        Thumbprint = $Thumbprint
+        Path = $Path
+    }
+}
+
+function Test-TargetResource 
+{
+    [CmdletBinding()]
+    [OutputType([Bool])]
+    param(
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-Thumbprint
+        } )]
+        [String]
+        $Thumbprint ,
+
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-CertificatePath
+        } )]
+        [String]
+        $Path ,
+
+        [Parameter()]
+        [ValidateSet(
+             'LocalMachine'
+        )]
+        [String]
+        $Location = 'LocalMachine' ,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Store = 'My'
+    )
+
+    [Bool](
+        'Cert:' | 
+        Join-Path -ChildPath $Location | 
+        Join-Path -ChildPath $Store | 
+        Get-ChildItem | 
+        Where-Object { $_.Thumbprint -ieq $Thumbprint }
+     )
+}
+
+function Set-TargetResource 
+{
+    [CmdletBinding(SupportsShouldProcess)]
+    param(
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-Thumbprint
+        } )]
+        [String]
+        $Thumbprint ,
+
+        [Parameter(
+            Mandatory
+        )]
+        [ValidateScript( {
+            $_ | Validate-CertificatePath
+        } )]
+        [String]
+        $Path ,
+
+        [Parameter()]
+        [ValidateSet(
+             'LocalMachine'
+        )]
+        [String]
+        $Location = 'LocalMachine' ,
+
+        [Parameter()]
+        [ValidateNotNullOrEmpty()]
+        [String]
+        $Store = 'My'
+    )
+
+    $certPath = 'Cert:' | Join-Path -ChildPath $Location | Join-Path -ChildPath $Store
+    if ($PSCmdlet.ShouldProcess("Importing certificate '$Path' into '$certPath'")) 
+    {
+        Import-Certificate -CertStoreLocation $certPath -FilePath $Path
+    }
+}
+
+Export-ModuleMember *-TargetResource
```

Comment:
should end a file with a new line per contribution guidelines.  [style guidelines - general rules #5](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)